subprojects/resources-s3/src/main/java/org/gradle/authentication/aws/AwsIamAuthentication.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.authentication.aws;
+
+import org.gradle.api.Incubating;
+import org.gradle.authentication.Authentication;
+
+/**
+ * Authentication scheme for digest access authentication over HTTP.
+ */
+@Incubating
+public interface AwsIamAuthentication extends Authentication {
```

Comment:
There's a typo in the class name. Should be `AwsImAuthentication` - `Im` as in "instance metadata". All other code would have to be aligned to follow the class name pattern.